cpp/core/tests/ExecutionCtxTest.cc

```diff
@@ -23,59 +23,104 @@ namespace gluten {
 
 class DummyExecutionCtx final : public ExecutionCtx {
  public:
-  std::shared_ptr<ResultIterator> getResultIterator(
+  ResourceHandle createResultIterator(
       MemoryManager* memoryManager,
       const std::string& spillDir,
       const std::vector<std::shared_ptr<ResultIterator>>& inputs,
       const std::unordered_map<std::string, std::string>& sessionConf) override {
     auto resIter = std::make_unique<DummyResultIterator>();
-    return std::make_shared<ResultIterator>(std::move(resIter));
+    auto iter = std::make_shared<ResultIterator>(std::move(resIter));
+    return resultIteratorHolder_.insert(iter);
   }
   MemoryManager* createMemoryManager(
       const std::string& name,
       std::shared_ptr<MemoryAllocator> ptr,
       std::unique_ptr<AllocationListener> uniquePtr) override {
     return nullptr;
   }
-  std::shared_ptr<ColumnarToRowConverter> getColumnar2RowConverter(MemoryManager* memoryManager) override {
+  ResourceHandle addResultIterator(std::shared_ptr<ResultIterator> ptr) override {
+    return 0;
```

Comment:
Does 0 have special meaning for a ResourceHandle ? If so we can create some const flags to avoid the magical uses.

Reply:
Just dummy implement for pass build, I'm ok for any changes here.

Reply:
Thanks! It was because I saw lots of `-1`s used in JniWrapper. Let's unify them all using a flag if possible.

Reply:
Done